bcdaworker/worker/worker.go

```diff
@@ -164,25 +168,64 @@ func (w *worker) ProcessJob(ctx context.Context, job models.Job, jobArgs models.
 	return nil
 }
 
-func moveFiles(tempDir string, stagingDir string) error {
+func compressFiles(ctx context.Context, tempDir string, stagingDir string) error {
+	logger := log.GetCtxLogger(ctx)
 	// Open the input file
 	files, err := os.ReadDir(tempDir)
 	if err != nil {
 		err = errors.Wrap(err, "Error reading from the staging directory for files for Job")
 		return err
 	}
+	gzipLevel, err := strconv.Atoi(os.Getenv("COMPRESSION_LEVEL"))
+	if err != nil || gzipLevel < 1 || gzipLevel > 9 { //levels 1-9 supported by BCDA.
+		gzipLevel = gzip.DefaultCompression
+		logger.Warnf("COMPRESSION_LEVEL not set to appropriate value.")
+	}
 	for _, f := range files {
 		oldPath := fmt.Sprintf("%s/%s", tempDir, f.Name())
 		newPath := fmt.Sprintf("%s/%s", stagingDir, f.Name())
-		err := os.Rename(oldPath, newPath) //#nosec G304
+		err := func() error {
```

Comment:
Can we add a comment here about why we have an anon func? Doesn't seem very intuitive at first glance 😄

Reply:
Added a comment, agree with you there. Seemed like the least-intensive way to ensure the defers run per-loop 😆 .